core/src/test/java/org/apache/iceberg/TestLocationProvider.java

```diff
@@ -285,4 +286,22 @@ public void testObjectStorageWithinTableLocation() {
     assertThat(parts).element(2).asString().isNotEmpty();
     assertThat(parts).element(3).asString().isEqualTo("test.parquet");
   }
+
+  @TestTemplate
+  public void testEncodedFieldNameInPartitionPath() {
```

Comment:
I feel like some form of this test should probably live in `TestPartitionPaths` as there are already some existing tests that specifically test for this

Reply:
I added a test in that class as well